### PR TITLE
UCT/IB: Use DV QP create_flags only if needed

### PR DESCRIPTION
**Pull request by Artemy-Mellanox**

### CODE REVIEW

File: src/uct/ib/rc/accel/rc_mlx5_common.c [hunk 1]
```diff
@@ -1027,13 +1027,13 @@ void uct_rc_mlx5_common_fill_dv_qp_attr(uct_rc_mlx5_iface_common_t *iface,
                                         unsigned scat2cqe_dir_mask)
 {
 #if HAVE_DECL_MLX5DV_QP_CREATE_ALLOW_SCATTER_TO_CQE
-    dv_attr->comp_mask   |= MLX5DV_QP_INIT_ATTR_MASK_QP_CREATE_FLAGS;
     dv_attr->create_flags = 0;
```
Comment (brminich) [inline src/uct/ib/rc/accel/rc_mlx5_common.c:1030]: do we need to delete setting flags to 0 now?

File: src/uct/ib/rc/accel/rc_mlx5_common.c [hunk 2]
```diff
@@ -1027,8 +1027,6 @@ void uct_rc_mlx5_common_fill_dv_qp_attr(uct_rc_mlx5_iface_common_t *iface,
                                         unsigned scat2cqe_dir_mask)
 {
 #if HAVE_DECL_MLX5DV_QP_CREATE_ALLOW_SCATTER_TO_CQE
-    dv_attr->create_flags = 0;
```
Comment (yosefe) [inline src/uct/ib/rc/accel/rc_mlx5_common.c:1031]: i guess it's ok because dv_attr zeroed out before the call

**Review by yosefe — APPROVED** [review]

**Review by brminich — DISMISSED** [review]

File: README.md [hunk 3]
```diff
@@ -131,6 +131,12 @@ $ make -C test/gtest test
   OS distribution or build glibc from source using versions less than 2.25 or
   greater than 2.29.
 
+* Due to compatibility flaw when using UCX with rdma-core v22 setting
+  UCX_DC_MLX5_RX_INLINE=0 is unsupported and will make DC transport unavailable.
+  This bug is fixed in rdma-core-22.4-2.el7.
+  See [ucx issue 5749](https://github.com/openucx/ucx/issues/5749) for more
+  details.
```
Comment (yosefe) [inline README.md:138]:
```
* Due to compatibility issue when using UCX with rdma-core v22, setting
  UCX_DC_MLX5_RX_INLINE=0 is unsupported and will make DC transport unavailable.
  This issue is fixed in rdma-core v22.4 and above.
  See [ucx issue 5749](https://github.com/openucx/ucx/issues/5749) for more
  details.
```

**Review by yosefe — APPROVED** [review]

**Comment by yosefe** [comment]
@Artemy-Mellanox can you pls port to v1.12.x if relevant?

**Pull request merged**